core/unit_test/TestHalfOperators.hpp

```diff
@@ -882,14 +882,17 @@ struct Functor_TestHalfOperators {
     }
 
     // TODO: figure out why operator{!,&&,||} are returning __nv_bool
-    actual_lhs(NEG)   = static_cast<double>(!h_lhs);
-    expected_lhs(NEG) = !d_lhs;
+    actual_lhs(NEG)   = static_cast<double>(!static_cast<bool>(h_lhs));
+    expected_lhs(NEG) = !static_cast<bool>(d_lhs);
 
-    actual_lhs(AND)   = static_cast<double>(half_type(0) && h_lhs);
-    expected_lhs(AND) = double(0) && d_lhs;
+    actual_lhs(AND) = static_cast<double>(static_cast<bool>(half_t(0)) &&
```

Comment:
revert this: purpose is to check implicit conversion to bool for use with && operator